code/go/0chain.net/smartcontract/zcnsc/burn.go

```diff
@@ -61,37 +60,30 @@ func (zcn *ZCNSmartContract) Burn(
 	if err != nil {
 		msg := fmt.Sprintf("payload decode error: %v, %s", err, info)
 		err = common.NewError(code, msg)
+		Logger.Error(msg, zap.Error(err))
 		return
 	}
 
 	if payload.EthereumAddress == "" {
-		err = common.NewError(code, "ethereum address is required "+info)
+		err = common.NewError(code, "ethereum address is required, "+info)
+		Logger.Error(err.Error(), zap.Error(err))
 		return
 	}
 
 	// get user node
 	un, err := GetUserNode(trans.ClientID, ctx)
-	switch err {
-	case nil:
-		if un.Nonce+1 != payload.Nonce {
-			err = common.NewError(
-				code,
-				fmt.Sprintf(
-					"nonce given (%v) for burning client (%s) must be greater by 1 than the current node nonce (%v) for Node.ID: '%s', %s",
-					payload.Nonce,
-					trans.ClientID,
-					un.Nonce,
-					un.ID,
-					info,
-				),
-			)
-			return
-		}
-	case util.ErrValueNotPresent:
-		err = common.NewError(code, "user node is nil "+info)
-		return
-	default:
+	if err != nil {
```

Comment:
not clear what has changed here

Reply:
I think this does not change the logic, it just a minor refactor to return early when err is not nil.